shared/lib/sharing/publicize/test/actions.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import nock from 'nock';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import Chai, { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	FETCH_PUBLICIZE_CONNECTIONS,
+	RECEIVE_PUBLICIZE_CONNECTIONS,
+	FAIL_PUBLICIZE_CONNECTIONS_REQUEST
+} from 'lib/redux-store/action-types';
+import {
+	fetchConnections,
+	receiveConnections,
+	failConnectionsRequest
+} from '../actions';
+
+/**
+ * Test setup
+ */
+Chai.use( sinonChai );
+
+describe( '#fetchConnections()', () => {
+	const spy = sinon.spy();
+
+	before( () => {
+		nock( 'https://public-api.wordpress.com:443' )
+			.persist()
+			.get( '/rest/v1.1/sites/2916284/publicize-connections' )
+			.reply( 200, { connections: [ { ID: 2, site_ID: 2916284 } ] } )
+			.get( '/rest/v1.1/sites/77203074/publicize-connections' )
+			.reply( 500 );
+	} );
+
+	beforeEach( () => {
+		spy.reset();
+	} );
+
+	after( () => {
+		nock.restore();
+	} );
+
+	it( 'should dispatch fetch action when thunk triggered', () => {
+		fetchConnections( 2916284 )( spy );
+
+		expect( spy ).to.have.been.calledWith( {
+			type: FETCH_PUBLICIZE_CONNECTIONS,
+			siteId: 2916284
+		} );
+	} );
+
+	it( 'should dispatch receive action when request completes', ( done ) => {
+		fetchConnections( 2916284 )( spy ).then( () => {
+			expect( spy ).to.have.been.calledTwice;
+
+			const action = spy.getCall( 1 ).args[ 0 ];
+			expect( action.type ).to.equal( RECEIVE_PUBLICIZE_CONNECTIONS );
+			expect( action.siteId ).to.equal( 2916284 );
+			expect( action.data.connections ).to.eql( [ { ID: 2, site_ID: 2916284 } ] );
```

Comment:
I agree that it is confusing to switch between tests and data structure definitions, but repeating the site ID is a bit magic-number-y to me.  Perhaps `TEST_SITE_ID` is worth defining as a constant, and nothing else.